src/Arn.ts

```diff
@@ -55,7 +55,7 @@ export class InvalidArnError extends Error {
 }
 
 function parseResourcePart(resourcePart: string): ArnResource {
-  const parts = resourcePart.match(/^([^:/]+)\/(.+)$/) || resourcePart.match(/^(.+?):(.+?):(.+)$/);
+  const parts = resourcePart.match(/^(.+?):(.+?):(.+)$/) || resourcePart.match(/^([^:\/]+)[\/|:](.+)$/);
```

Comment:
This regexp will allow the presence of a `|` character separating the  type and the id, won't it? Was this intentional or did you intend to use `|` as an "or" (which doesn't work inside `[ ]` brackets)?